### PR TITLE
[8pt] PR: Synthesize testcases hotfix for GVAL memory/performance issues

### PR DESCRIPTION
**Pull request by GregoryPetrochenkov-NOAA**
This pull request addresses memory and performance issues associated with the first integration of gval in synthesize test cases.  #973 

### Changes
- tools/tools_shared_functions.py:  Remove mask_and_scale, style conformance
- Pipfile: Update gval version
- pyproject.toml:  precommit update
- tools/run_test_case.py: style conformance
- tools/synthesize_test_cases.py: refactor changes in and make traceback
- tools/inundate_mosaic_wrapper:  Check for string type for huc list

### Testing
Local unit testing as well as comparison of old and new metrics csv

### Issuer Checklist (For developer use)

_You may update this checklist before and/or after creating the PR. If you're unsure about any of them, please ask, we're here to help! These items are what we are going to look for before merging your code._

- [x] Informative and human-readable title, using the format: `[_pt] PR: <description>`
- [x] Links are provided if this PR resolves an issue, or depends on another other PR
- [x] If submitting a PR to the `dev` branch (the default branch), you have a descriptive [Feature Branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) name using the format: `dev-<description-of-change>` (e.g. `dev-revise-levee-masking`)
- [x] Changes are limited to a single goal (no scope creep)
- [x] The feature branch you're submitting as a PR is up to date (merged) with the latest `dev` branch
- [x] `pre-commit` hooks were run locally
- [x] Any _change_ in functionality is tested
- [x] Passes all unit tests locally (inside interactive Docker container, at `/foss_fim/`, run: `pytest unit_tests/`)
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] [CHANGELOG](/docs/CHANGELOG.md) updated with template version number, e.g. `4.x.x.x`
- [x] [Reviewers requested](https://help.github.com/articles/requesting-a-pull-request-review/)
- [x] Add yourself as an [assignee](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) in the PR  as well as the FIM Technical Lead

### Merge Checklist (For Technical Lead use only)

- [ ] Update [CHANGELOG](/docs/CHANGELOG.md) with latest version number and merge date
- [ ] Update the [Citation.cff](/CITATION.cff) file to reflect the latest version number in the [CHANGELOG](/docs/CHANGELOG.md)
- [ ] If applicable, update [README](/README.md) with major alterations

### CODE REVIEW

**Comment by GregoryPetrochenkov-NOAA** [comment]
Attached is the list of huc and magnitude combindations with different nws_lids.  @CarsonPruitt-NOAA [
[nws_lid_diff.csv](https://github.com/NOAA-OWP/inundation-mapping/files/12896392/nws_lid_diff.csv)

**Comment by GregoryPetrochenkov-NOAA** [comment]
There is 100% agreement now.

File: src/utils/shared_functions.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import glob
 import inspect
 import os
+import traceback
```
Comment (robgpita-noaa) [inline src/utils/shared_functions.py:6]: @GregoryPetrochenkov-NOAA is the `traceback` module needed here?

File: tools/inundate_mosaic_wrapper.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import argparse
 import errno
 import os
+import string
```
Comment (robgpita-noaa) [inline tools/inundate_mosaic_wrapper.py:4]: This could be removed as well.

**Review by robgpita-noaa — APPROVED** [review]
@GregoryPetrochenkov-NOAA Thanks for all of your work on this. 

I was successfully able to run `synthesize_test_cases.py` on a BED run with the new  docker container (updated GVAL version) and the changes proposed here. 

This PR also passes the git-hooks, as well as passes all of the unit tests which are currently working.

File: src/utils/shared_functions.py (hunk 1 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/utils/shared_functions.py]: I went ahead and removed this import

File: tools/inundate_mosaic_wrapper.py (hunk 2 above)
Comment (GregoryPetrochenkov-NOAA) [reply tools/inundate_mosaic_wrapper.py]: I went ahead and removed this import

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,19 @@ Revise stream clipping to WBD by (1) reducing the buffer to clip streams away fr
 
 <br/><br/>
 
+## v4.4.2.4 - 2023-10-11 - [PR#1007](https://github.com/NOAA-OWP/inundation-mapping/pull/1007)
```
Comment (mluck) [inline docs/CHANGELOG.md:18]: Minor formatting suggestions: Make the version number `v4.x.x.x` and move this PR to the top of the CHANGELOG. Add `<br/><br/>` at the end of this PR preceded and followed by a blank line.

**Review by mluck — APPROVED** [review]
Looks good. I just left a minor comment on formatting in the CHANGELOG.

File: docs/CHANGELOG.md (hunk 3 above)
Comment (GregoryPetrochenkov-NOAA) [reply docs/CHANGELOG.md:18]: I went ahead and added the new line under the CHANGELOG.MD entry.

**Review by robgpita-noaa — DISMISSED** [review]

**Review by CarsonPruitt-NOAA — DISMISSED** [review]

**Review by CarsonPruitt-NOAA — APPROVED** [review]

**Pull request merged**